x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/20_semantic_text_field_mapper.yml

```diff
@@ -50,19 +51,109 @@ setup:
                 type: text
 
 ---
-"Sparse vector results format":
+"Dense vector results are indexed as nested chunks and searchable":
+    - do:
+        bulk:
+          index: test-index
+          refresh: true
+          body: |
+            {"index":{}}
+            {"dense_field": ["you know, for testing", "now with chunks"]}
+            {"index":{}}
+            {"dense_field": ["some more tests", "that include chunks"]}
+
     - do:
-        index:
+        search:
           index: test-index
-          id: doc_1
           body:
-            sparse_field: "you know, for testing"
+            query:
+              nested:
+                path: dense_field.inference.chunks
+                query:
+                  knn:
+                    field: dense_field.inference.chunks.embeddings
+                    query_vector_builder:
+                      text_embedding:
+                        model_id: dense-inference-id
+                        model_text: "you know, for testing"
+
+    - match: { hits.total.value: 2 }
+    - match: { hits.total.relation: eq }
+
+    # Search with inner hits
+    - do:
+        search:
+          _source: false
+          index: test-index
+          body:
+            query:
+              nested:
+                path: dense_field.inference.chunks
+                inner_hits:
+                  _source: false
+                  fields: [dense_field.inference.chunks.text]
+                query:
+                  knn:
+                    field: dense_field.inference.chunks.embeddings
+                    query_vector_builder:
+                      text_embedding:
+                        model_id: dense-inference-id
+                        model_text: "you know, for testing"
+
+    - match: { hits.total.value: 2 }
+    - match: { hits.total.relation: eq }
+    - match: { hits.hits.0.inner_hits.dense_field\.inference\.chunks.hits.total.value: 2 }
+    - exists:  hits.hits.0.inner_hits.dense_field\.inference\.chunks.hits.hits.0.fields.dense_field\.inference\.chunks.0.text
```

Comment:
We should be able to match the exact text of the chunk?

Reply:
Sorry @jimczi  - merged already. But I've added this as part of #107262 now 👍